net/portmapper/portmapper.go

```diff
@@ -66,15 +65,36 @@ type Client struct {
 	pcpSawTime  time.Time // time we last saw PCP was available
 	uPnPSawTime time.Time // time we last saw UPnP was available
 
-	localPort  uint16
-	pmpMapping *pmpMapping // non-nil if we have a PMP mapping
+	localPort uint16
+
+	mapping mapping // non-nil if we have a mapping
+}
+
+// mapping represents a created port-mapping over some protocol.  It specifies a lease duration,
+// how to release the mapping, and whether the map is still valid.
+//
+// After a mapping is created, it should be immutable, and thus reads should be safe across
+// concurrent goroutines.
+type mapping interface {
+	// isCurrent reports whether a mapping is valid now, since most mappings have a lease.
+	// isCurrent() bool
+	// release will attempt to unmap the established port mapping. Will block until completion,
+	// but can be called asynchronously. release should be idempotent, and thus even if called
+	// multiple times should not cause additional side-effects.
+	release(context.Context)
```

Comment:
why's this one unexported but the rest are capitalized?

Normally you make one thing unexported as a way to prevent people from outside the package from implementing it, like:

https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/testing/testing.go;l=629

(see that `private()` method)

But that's not what's happening here. Seems like this should be exported?

Reply:
to be honest, was super lazy here and just made the others exported so it wouldn't conflict with field names. I can export it though